bitnami/mongodb/templates/replicaset/scripts-configmap.yaml

```diff
@@ -254,28 +254,34 @@ data:
       # read rs.conf again and store it. settings format is '"<key>" : <value>,'
       currentRsConf=$(mongosh ${usernameAndPassword} --eval 'rs.conf()')
 
-      desiredEqualsactual=unknown
+      desiredEqualsActual=unknown
       settingsToConfigure=""
       for key in ${!desiredRsConf[@]}; do
         value=${desiredRsConf[$key]}
-        if ! $(echo "\"${currentRsConf}"\" | grep -q -e "${key}: ${value},"); then
-           if [[ $key =~ ^members\[[0-9]+\]\..+ ]]; then
-            memberIndex=$(echo $key | grep -o -E '[0-9]+')
-            nodeConfigKey=${key#*.}
-            settingsToConfigure="${settingsToConfigure}cfg.members[${memberIndex}].${nodeConfigKey} = ${value}; "
+        if [[ $key =~ ^members\[[0-9]+\]\..+ ]]; then
+          # Replica set member specific setting
+          if "$(mongosh "cfg=${currentRsConf}; cfg.${key}" 2>/dev/null)" != "${value}"; then
```

Comment:
Sorry @jkraml-staffbase  I don't fully understand this line. Could you explain it? I see these errors when I try to execute it:
```console
MongoshInvalidInputError: [COMMON-10001] Invalid URI: cfg={....}
```

Reply:
@fmulero sorry, I must've made a cop&paste mistake between what I tested in our cluster and what I committed. This line is missing `--eval`.

The idea is that we want to be able to evaluate the config which is given as e.g. `members[1].priority = 1`. The script recognizes `members[1].priority` as the key of a settings, but the old  grep variant would not be able to find this in the output (which was `{…,"members":[{"priority": 1, …}]}`.

We use `mongosh` as a javascript interpreter here.

The expression sets `cfg` to the previously retrieved value if `rs.conf()` and then tries to extract the desired value by replacing `cfg.${key}`, giving for example `cfg.members[0].priority`, which would be a valid javascript expression.

I'll push a fixed commit right away.

Reply:
(sorry I didn't reply earlier, Firefox apparently does not refresh the tab on restart, so I didn't see your comment :laughing:)

Reply:
👍